### DISCUSSION TITLE
Shouldn't CategoricalCnv be based on a CanonicalLocation vs a Contextual Location?

### DISCUSSION THREAD

**Discussion by larrybabb** [ga4gh/cat-vrs#17] [Q&A]
https://github.com/ga4gh/cat-vrs/blob/e5f5cbd7ba3e42a9846441a930f3e76ba2995fb5/schema/catvrs/catvrs-source.yaml#L115

Currently, the CategoricalCnv is defined by a vrs `Location` which is defined by a region on an exact sequence (aka a contextual location).  This means that if you want to define a CategoricalCnv to be based on a region that spans different assemblies, you can't - at least not in this representation.

This class was assembled and not very well discussed and defined up to this point. While I understand the idea of using an optional 'locationMatch' attribute to provide some guidance to a user of the CategoricalCnv to understand how the author of it meant for it to be used in matching assayed CNVs in the wild, I'm not sure this is well thought through.

I think we need to find a way to both 1) define a canonical location so that cnvs are limited to one specific assembly and 2) find a method for allowing the author of the Categorical CNV to define the scope of it in terms of including observed CNVs from assays.

**Comment by DanielPuthawala** [comment]
I think this sounds like a great thing to hash out with you, me and Alex sometime soon.  I had just assumed that what was inherited from VRSitile was what you guys were happy with, but if not, I reckon it's good to get that ironed out sooner than later.  Do you and @ahwagner  already have any scheduled 1-on-1s coming up that I can crash so we can resolve this?

**Comment by DanielPuthawala** [comment]
But the desire to use something like DefiningContext instead of Location makes sense to me...

**Comment by ahwagner** [comment]
Hey @larrybabb, here's what we were thinking under the constraint model (following [this general pattern](https://github.com/ga4gh/cat-vrs/discussions/22?sort=top#discussioncomment-9547661) for categorical variants):

![Cat-VRS Constraint Model](https://github.com/ga4gh/cat-vrs/assets/811065/532a500e-c2e6-4bad-99d6-9e8fb663e460)

In this mock-up, the defining context gives us the "root" from which liftovers are valid, and the relation that says liftover operations are what is expected (and not homologous sequences in other organisms, etc.). It also provides, though _members_, a list of precise locations that the instance pre-computed liftovers for. IMO, this reflects the reality that any categorical variant located on a reference has an originating context, from which additional contexts may be computed. Does this capture what you were thinking of? If not, what gaps do you see?

**Reply by larrybabb** [reply]
That looks fine. See my comments in the general discussion for my concerns about how to roll this out.

If you could demo the Canonical Allele in a constraint diagram that would be much more helpful.

**Reply by DanielPuthawala** [reply]
Got demos up for you [in the other thread](https://github.com/ga4gh/cat-vrs/discussions/22)!
